possible Bug: app crashes when invoking settings of a DecSync CC calendar

When tapping on a calendar from [DecSync CC](https://github.com/39aldo39/DecSyncCC) in the settings of Etar the app crashes. So it is not possible to change the settings of that calendar in Etar.
It does not happen with a calender from another app.

I'm not sure if this is a issue with Etar or DecSync CC. If the latter is the case please let me know so I can create an issue there.

Same here, running LineageOS 18.1 on a OnePlus 3. However, it not only occurs with all of my _DecSync_ calendars but also with the one from [_Birthday Adapter_](https://f-droid.org/de/packages/org.birthdayadapter/). Also checked with the F-Droid release of Etar, same result.

Stack trace provided by [_Scoop_](https://f-droid.org/de/packages/taco.scoop/):

> FATAL EXCEPTION: main
> Process: org.lineageos.etar, PID: 19405
> java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
> 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602)
> 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
> Caused by: java.lang.reflect.InvocationTargetException
> 	at java.lang.reflect.Method.invoke(Native Method)
> 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
> 	... 1 more
> Caused by: android.content.pm.PackageManager$NameNotFoundException: org.decsync.cc
> 	at android.app.ApplicationPackageManager.getApplicationInfoAsUser(ApplicationPackageManager.java:423)
> 	at android.app.ApplicationPackageManager.getApplicationInfo(ApplicationPackageManager.java:412)
> 	at android.app.ApplicationPackageManager.getResourcesForApplication(ApplicationPackageManager.java:1688)
> 	at com.android.calendar.settings.CalendarPreferences.getAuthenticatorInfo(CalendarPreferences.kt:196)
> 	at com.android.calendar.settings.CalendarPreferences.populatePreferences(CalendarPreferences.kt:68)
> 	at com.android.calendar.settings.CalendarPreferences.onCreatePreferences(CalendarPreferences.kt:53)
> 	at androidx.preference.PreferenceFragmentCompat.onCreate(PreferenceFragmentCompat.java:161)
> 	at androidx.fragment.app.Fragment.performCreate(Fragment.java:2883)
> 	at androidx.fragment.app.FragmentStateManager.create(FragmentStateManager.java:484)
> 	at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1312)
> 	at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1477)
> 	at androidx.fragment.app.FragmentTransition.addToFirstInLastOut(FragmentTransition.java:1246)
> 	at androidx.fragment.app.FragmentTransition.calculateFragments(FragmentTransition.java:1128)
> 	at androidx.fragment.app.FragmentTransition.startTransitions(FragmentTransition.java:135)
> 	at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:2112)
> 	at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:2055)
> 	at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:1957)
> 	at androidx.fragment.app.FragmentManager$5.run(FragmentManager.java:496)
> 	at android.os.Handler.handleCallback(Handler.java:938)
> 	at android.os.Handler.dispatchMessage(Handler.java:99)
> 	at android.os.Looper.loop(Looper.java:223)
> 	at android.app.ActivityThread.main(ActivityThread.java:7664)
> 	... 3 more

--- a/AndroidManifest.xml
+++ b/AndroidManifest.xml
@@ -35,6 +35,8 @@
         <package android:name="at.bitfire.davdroid" />
         <package android:name="com.etesync.syncadapter" />
         <package android:name="com.google.android.gm" />
+        <package android:name="org.birthdayadapter" />
+        <package android:name="org.decsync.cc" />
 
         <intent>
             <action android:name="android.intent.action.VIEW" />

--- a/src/com/android/calendar/settings/CalendarPreferences.kt
+++ b/src/com/android/calendar/settings/CalendarPreferences.kt
@@ -21,6 +21,7 @@ import android.accounts.Account
 import android.accounts.AccountManager
 import android.accounts.AuthenticatorDescription
 import android.content.Intent
+import android.content.pm.PackageManager
 import android.graphics.drawable.Drawable
 import android.os.Bundle
 import android.provider.CalendarContract
@@ -196,13 +197,22 @@ class CalendarPreferences : PreferenceFragmentCompat() {
     private fun getAuthenticatorInfo(account: Account): AuthenticatorInfo? {
         val description = getAuthenticatorDescription(account) ?: return null
 
-        val pm = activity?.packageManager
-        val label = pm?.getResourcesForApplication(description.packageName)?.getString(
+        return try {
+            val pm = activity?.packageManager
+            val label = pm?.getResourcesForApplication(description.packageName)?.getString(
                 description.labelId)
-        val icon = pm?.getDrawable(description.packageName, description.iconId, null)
-        val intent = pm?.getLaunchIntentForPackage(description.packageName)
+            val icon = pm?.getDrawable(description.packageName, description.iconId, null)
+            val intent = pm?.getLaunchIntentForPackage(description.packageName)
 
-        return AuthenticatorInfo(label, icon, intent)
+            AuthenticatorInfo(label, icon, intent)
+
+        } catch (e: PackageManager.NameNotFoundException) {
+            val errorDialog = AlertDialog.Builder(requireActivity())
+                .setMessage("$e")
+                .create()
+            errorDialog.show()
+            return null
+        }
     }
 
     private fun getAuthenticatorDescription(account: Account): AuthenticatorDescription? {

